### PR TITLE
imp: add default methods for writeValueAsString

### PR DESCRIPTION
**Pull request by sdelamo**
To provide an API such as `com.fasterxml.jackson.databind.ObjectMapper::writeValueAsString`

### CODE REVIEW

File: json-core/src/main/java/io/micronaut/json/JsonMapper.java [hunk 1]
```diff
@@ -201,6 +202,29 @@ default Processor<byte[], JsonNode> createReactiveParser(@NonNull Consumer<Proce
...
+
+    /**
+     * Write an object as String.
+     *
+     * @param object The object to serialize.
+     * @param  charset The {@linkplain java.nio.charset.Charset charset} to be used to decode the {@code bytes}.
+     * @return The serialized encoded json.
+     * @throws IOException IOException
+     */
+    default String writeValueAsString(@Nullable Object object, @NonNull Charset charset) throws IOException {
```
Comment (yawkat) [inline json-core/src/main/java/io/micronaut/json/JsonMapper.java]: remove this, jackson doesnt have it, nobody will use it, and it doesnt make sense for json

Comment (sdelamo) [reply json-core/src/main/java/io/micronaut/json/JsonMapper.java]: resolved via https://github.com/micronaut-projects/micronaut-core/pull/9402/commits/8f59667c2da700ad9bcfd4bc4f2d80924e76ab6c.

**Comment by sdelamo** [comment]
I have added a `readValue(String, Class)` as well.

**Review by yawkat — APPROVED** [review]

**Review by graemerocher — APPROVED** [review]

**Comment by sdelamo** [comment]
The failure seems unrelated. Investigating

**Pull request merged**